src/plugins/cesium/ngv-plugin-cesium-navigation.ts

```diff
@@ -0,0 +1,320 @@
+    }
+
+    .view-btns > div {
+      display: flex;
+      flex-direction: column;
+      row-gap: 5px;
+    }
+
+    .view-container {
+      display: flex;
+      flex-direction: column;
+      row-gap: 5px;
+      padding: 5px;
+    }
+
+    .view-container > h4 {
+      margin: 0;
+      padding: 5px;
+    }
+
+    .view-container > h4:hover {
+      background-color: lightyellow;
+      cursor: pointer;
+    }
+
+    .divider {
+      width: 100%;
+      border: 1px solid #e0e3e6;
+    }
+  `;
+
+  parseViews(): void {
+    this.views = this.viewsConfig.map((v) => {
+      const positions: Cartographic[] = v.positions.map((p) =>
+        Cartographic.fromDegrees(p[0], p[1]),
+      );
+
+      const baseRect = Rectangle.fromCartographicArray(positions);
+
+      const nw = Cartographic.toCartesian(Rectangle.northwest(baseRect));
+      const ne = Cartographic.toCartesian(Rectangle.northeast(baseRect));
+      const sw = Cartographic.toCartesian(Rectangle.southwest(baseRect));
+      const se = Cartographic.toCartesian(Rectangle.southeast(baseRect));
+      const bottomPositions = [nw, ne, se, sw];
+      updateHeightForCartesianPositions(
+        bottomPositions,
+        v.elevation,
+        this.viewer.scene,
+        true,
+      );
+      const [nwt, net, set, swt] = updateHeightForCartesianPositions(
+        bottomPositions,
+        v.elevation + v.height,
+        this.viewer.scene,
+      );
+
+      const highlightEntity = this.dataSource.entities.add(
+        new Entity({
+          show: false,
+          polygon: {
+            hierarchy: bottomPositions,
+            heightReference: HeightReference.RELATIVE_TO_GROUND,
+            extrudedHeightReference: HeightReference.RELATIVE_TO_GROUND,
+            material: v.highlightColor
+              ? Color.fromCssColorString(v.highlightColor)
+              : Color.RED.withAlpha(0.6),
+            extrudedHeight: v.height,
+          },
+        }),
+      );
+
+      return {
+        top: {
+          ...calculateViewOnRectangle(swt, set, nwt, net, v.fovAngle),
+          duration: v.flyDuration,
+        },
+        north: {
+          ...calculateViewOnRectangle(ne, nw, net, nwt, v.fovAngle),
+          duration: v.flyDuration,
+        },
+        east: {
+          ...calculateViewOnRectangle(se, ne, set, net, v.fovAngle),
+          duration: v.flyDuration,
+        },
+        west: {
+          ...calculateViewOnRectangle(nw, sw, nwt, swt, v.fovAngle),
+          duration: v.flyDuration,
+        },
+        south: {
+          ...calculateViewOnRectangle(sw, se, swt, set, v.fovAngle),
+          duration: v.flyDuration,
+        },
+        title: v.title,
+        highlightEntity,
+      };
+    });
+  }
+
+  willUpdate(changedProperties: PropertyValues): void {
+    if (changedProperties.has('currentViewIndex')) {
+      this.currentView = this.views[this.currentViewIndex];
+      this.viewer.camera.flyTo(this.currentView.south);
+    }
```

Comment:
Let's use a Task instead of testing strings. It will avoid bugs if we rename the property.